test/features/support/env.js

```diff
@@ -14,46 +15,271 @@
             });
         });
     });
-};
 
-module.exports = configure;
\ No newline at end of file
+    Before(function () {
+        const world = this;
+
+        // Make sure this is ran only once.
+        if(!world.user) {
+            world.user = null;
+        }
+
+        return browser.addMockModule('firebase', function () {
+
+            angular.module('firebase').factory('$firebaseObject', function () {
+                return function () {
+                    return {
+                        $loaded: function () {
+                            return Promise.resolve({});
+                        }
+                    };
+                };
+            });
+
+            angular.module('firebase').factory('$firebaseArray', function () {
+                return function () {
+                    return {
+                        $loaded: function () {
+                            return Promise.resolve({});
+                        }
+                    };
+                };
+            });
+
+             angular.module('firebase').factory('$firebaseAuth', function () {
+
+                 window.firebase = {
+                     initializeApp: function () {
+                         return Promise.resolve();
+                     },
+                     database: function () {
+                        return {
+                            ref: function () {
+                                return Promise.resolve();
+                            }
+                        };
+                     }
+                 };
+
+                    return function () {
+                        return {
+                            $waitForSignIn: function () {
+                                return Promise.resolve();
+                            },
+                            $getAuth: function () {
+                                return null;
+                            },
+                            $onAuthStateChanged: function () {
+
+                            },
+                            $createUserWithEmailAndPassword: function () {
+                                return Promise.resolve({
+                                    sendEmailVerification: function () {
+                                        return Promise.resolve();
+                                    }
+                                });
+                            },
+                            $signInWithEmailAndPassword: function () {
+                                return Promise.resolve({
+                                    emailVerified: true
+                                });
+                            },
+                            $signOut: function () {
+
+                                return Promise.resolve();
+                            }
+                        };
+                    };
+                }
+            );
+        });
+
+    });
+
+    Before("@mock-user-logged-in", function () {
```

Comment:
Move mocks for auth/firebase to own support files. For example `support/firebase.js`